main.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+)
+
+type Entry struct {
+	ID          uint
+	FirstName   string
+	LastName    string
+	PhoneNumber string
+}
+
+func mainMenu() (choice int) {
+	fmt.Println(`Choose your action:
+1) List all entries.
+2) Add new entry.
+3) Remove an entry by ID.
+4) Quit`)
+	fmt.Scanln(&choice)
+	return choice
+}
+
+func openFile() []Entry {
+	var entries []Entry
+	file, err := os.Open("phonebook.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	if err := json.NewDecoder(bufio.NewReader(file)).Decode(&entries); err != nil {
+		log.Fatal(err)
+	}
+	return entries
+}
+
+func listEntries() {
+	entries := openFile()
+	sort.Slice(entries, func(i, j int) bool { return entries[i].LastName < entries[j].LastName })
+	fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+	fmt.Println("-------------------------------------------------------------------------")
+	for i, e := range entries {
+		fmt.Printf("%10d %20s %20s %20s\n", e.ID, e.LastName, e.FirstName, e.PhoneNumber)
+		if (i+1)%20 == 0 && i < len(entries)-1 {
+			fmt.Print("Press <ENTER> to continue...")
+			fmt.Scanln()
+			fmt.Printf("%10s %20s %20s %20s\n", "ID", "Last Name", "First Name", "Phone Number")
+			fmt.Println("-------------------------------------------------------------------------")
+		}
+	}
+}
+
+func addNew() {
+	var ID uint
+	entries := openFile()
+	for i, e := range entries {
+		if i == len(entries)-1 {
```

Comment:
This isn't entirely correct. You need to find max ID not the ID of the last element (which would be as simple as `entries[len(entries)-1].ID` w/o the loop)